src/js/common/schemaform/routing.js

```diff
@@ -7,12 +7,12 @@ import { expandArrayPages } from './helpers';
  * their dependencies and therefore should be skipped.
  */
 function getEligiblePages(pageList, data, pathname) {
-  const eligiblePageList = getActivePages(pageList, data);
   // Any `showPagePerItem` pages are expanded to create items for each array item.
   // We update the `path` for each of those pages to replace `:index` with the current item index.
-  const expandedPageList = expandArrayPages(eligiblePageList, data);
+  const expandedPageList = expandArrayPages(pageList, data);
   const pageIndex = _.findIndex(item => item.path === pathname, expandedPageList);
-  return { pages: expandedPageList, pageIndex };
+  const eligiblePageList = getActivePages(expandedPageList, data);
```

Comment:
I think this needs to be above the line that gets `pageIndex`. Otherwise, we might get the wrong index if some pages are filtered out in the this step?

Reply:
:thinking: Yeah, good catch.